<link rel="stylesheet" href="styles.css">
<script src="./script.js"></script>

<div class="navbar">
  <span class="prev"><a href="../main.html">Notes</a></span>
  <span class="center"><a href="#">Programming questions - Exceptional Handling</a></span>
  <span class="next"><a href="./00.main.html">Python - Interview Prep</a></span>
</div>

<div class="main-content">
  <div class="content">
    <div class="topic" id="ReturnInTryWithFinally">
      <div class="topic-heading"><h2>Return in Try with Finally</h2></div>
      <div class="topic-body">
        <pre><code class="language-python">
    def foo():
        try:
            return 1
        finally:
            return 2
    k = foo()
    print(k)  # 1
        </code></pre>
      </div>
    </div>

    <hr/>
    
    <div class="topic" id="TryExceptElseFinally">
      <div class="topic-heading"><h2><span class="code">try</span>, <span class="code">except</span>, <span class="code">else</span>, <span class="code">finally</span></h2></div>
      <div class="topic-body">
        <pre><code class="language-python">
    def divide(x, y):
        try:
            result = x / y
        except ZeroDivisionError:
            print("division by zero!")
        else:
            print("result is", result)
        finally:
            print("executing finally clause")

    divide(2, 1)            # Output: result is 2.0 (executing finally clause)
    divide(2, 0)            # Output: division by zero! (executing finally clause)
    divide("2", "1")        # Output: executing finally clause (Then a TypeError is raised)
        </code></pre>
      </div>
    </div>

    <hr/>

  </div>

  <div class="sidebar-right">
    <div class="menu-header" id="menu-header">Table of Content</div>
    <div class="menu-content" id="menu-content"></div>
  </div>

</div>

<div class="footer">
  <span class="prev"><a href="./05.event_handling.html">05. Event Handling</a></span>
  <span class="next"><a href="./07.lists_and_keys.html">07. Lists and Keys</a></span>
</div>
